### PR TITLE
HAWQ-145. Memory leak when free the argument instance of resource lea…

### PR DESCRIPTION
**Pull request by jiny2**
This is to fix a memory leak.

### CODE REVIEW

File: src/backend/resourcemanager/communication/rmcomm_QD2RM.c [hunk 1]
```diff
@@ -1438,6 +1438,7 @@ void freeHeartBeatThreadArg(HeartBeatThreadArg *arg)
 	{
 		free((*arg)->HostAddrs[i]);
 	}
+	free((*arg)->HostAddrs);
```
Comment (zhangh43) [inline src/backend/resourcemanager/communication/rmcomm_QD2RM.c]: have a look at the code: tharg->HostAddrs = malloc(sizeof(char *) * tharg->HostAddrSize);
where tharg->HostAddrSize maybe zero and malloc(0) happens.
what i mean yesterday is to test whether tharg->HostAddrSize > 0 before free arg->HostAddrs

Comment (zhangh43) [inline src/backend/resourcemanager/communication/rmcomm_QD2RM.c]: sorry. It's OK to free(malloc(0)).
+1

**Comment by linwen** [comment]
+1

File: src/backend/resourcemanager/communication/rmcomm_QD2RM.c (hunk 1 above)
Comment (jiny2) [inline src/backend/resourcemanager/communication/rmcomm_QD2RM.c]: In the code, I have logic to prevent this case :)

**Pull request closed without merging**